src/components/CompleteProfile/CompleteProfile.tsx

```diff
@@ -1,44 +1,154 @@
-	id: string;
-	name: string;
+interface ProfileParams {
+	email: string;
+	isBasedOnGTA: boolean;
+	canJoinLocalEvents: boolean;
+	name?: string;
+	links: { platform: string, url: string }[] | [];
+	pronouns?: string;
+	socialAccounts?: string[];
+	birthday: string;
+	avatar?: string;
+	skills: string[];
 }
 
+type UpdateProfileParams = Omit<ProfileParams, 'email'>;
+
+const platformEnum = ['slack', 'linkedin', 'github', 'portfolio', 'instagram', 'threads', 'facebook', 'bluesky', 'mastodon', 'xtwitter', 'dev'];
+
+// Validation helpers
+/**
+ * Checks whether a birthday string is in the valid MM-DD format.
+ *
+ * A valid format must follow the pattern: two-digit month (01–12)
+ * followed by a dash and a two-digit day (01–31).
+ * This function does not validate actual calendar dates (e.g., "02-30" would be considered valid).
+ *
+ * @param birthday - The birthday string to validate, expected in MM-DD format.
+ * @returns `true` if the format is valid, otherwise `false`.
+ */
+const isValidBirthdayFormat = (birthday: string): boolean => {
+	const regex = /^(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01])$/;
+	return regex.test(birthday);
+};
+
+/**
+ * Determines whether a birthday string represents a real calendar date.
+ *
+ * The input must be in MM-DD format and represent a valid calendar day.
+ * This function uses a placeholder year (2000, a leap year) to account for month lengths,
+ * including February 29th. It assumes no year context is needed and is used primarily for validation.
+ *
+ * @deprecated We will remove update this when Temporal becomes stable
+ *
+ * @param birthday - The birthday string to validate, expected in MM-DD format.
+ * @returns `true` if the string is a valid date in the MM-DD format, otherwise `false`.
+ */
+const isRealDate = (birthday: string): boolean => {
+	if (!isValidBirthdayFormat(birthday)) { return false; }
+
+	const [monthStr, dayStr] = birthday.split('-');
+	const month = Number(monthStr);
+	const day = Number(dayStr);
+
+	const date = new Date(2000, month - 1, day);
+	return date.getMonth() + 1 === month && date.getDate() === day;
+};
+
+const updateProfile = async (data: UpdateProfileParams, profileId: string) => {
+	try {
+		const response = await fetch(`/api/profiles/${profileId}`, {
+			method: 'PATCH',
+			headers: {
+				'Content-Type': 'application/json'
+			},
+			body: JSON.stringify({ ...data }),
+			credentials: 'include'
+		});
+		if (response.status === 200) {
+			window.location.href = '/pages/home/';
+		} else {
+			const errorData = await response.json();
+			// TODO: Replace console log with error message once the design becomes available
+			console.log('Response not ok: ', errorData ?? 'An Error Occurred!');
+		}
+	} catch (e) {
+		if (import.meta.env.MODE === 'development') {
+			if (e instanceof Error) {
+				console.error(e.name);
+				console.error(e.cause);
+				console.error(e.message);
+				console.error(e.stack);
+			} else {
+				throw new Error('Error Encountered!');
```

Comment:
```suggestion
				console.log(e);
				throw new Error('Unknown Error Encountered!');
```
@kbventures  I can do the above for more clarity if required.